pygmt/tests/test_plot.py

```diff
@@ -295,6 +299,25 @@ def test_plot_sizes_colors_transparencies():
     return fig
 
 
+@pytest.mark.mpl_image_compare
+def test_plot_symbols(data, region):
+    """
+    Plot the data using array-like symbols.
+    """
+    fig = Figure()
+    fig.plot(
+        x=data[:, 0],
+        y=data[:, 1],
+        region=region,
+        projection="X4c",
+        color="blue",
+        sizes=np.full(len(data[:, 0]), 0.5),
+        symbol=np.full(len(data[:, 0]), "c"),
```

Comment:
At the moment, this test is checking that plotting all symbols as circles (c) work. A better test would be to test a mix of symbols such as circles (c), triangles (t), squares (s), etc (like your example in https://github.com/GenericMappingTools/pygmt/issues/1076#issue-834922951).